src/util/compat.hpp

```diff
@@ -34,4 +34,27 @@ inline void compiler_barrier_if_xlc_leq(unsigned ver) {
 template <typename X>
 inline constexpr bool isinf(X x) { return std::isinf(x); }
 
+// Work around a bad inlining-related optimization with icpc 16.0.3 and -xMIC-AVX512,
+// by forcing a computation.
+
+template <typename X>
+inline void sink(const X& x) {
```

Comment:
This look like a lot of compute. Is this the easiest way?
http://stackoverflow.com/questions/2219829/how-to-prevent-gcc-optimizing-some-statements-in-c
I'm way out of my depth here, so please ignore.

Just to be sure how often is this called?